src/core/digest.rs

```diff
@@ -0,0 +1,54 @@
+use serde::{Deserialize, Serialize};
+
+/// Generic Header Digests
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct Digests {
```

Comment:
Hmm what are these digests? Also, couldn't we use `Vec<Digest>` instead of a new `Digests` structure? Out of curiosity.